src/main/java/org/spring/FullVersion/Family.java

```diff
@@ -0,0 +1,120 @@
+package org.spring.FullVersion;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class Family {
+
+    private Human mother;
+    private Human father;
+    private Human[] children;
+    private Pet pet;
+
+    public Family() {
+    }
+
+    public Family(Human father, Human mother) {
+        this.father = father;
+        this.mother = mother;
+        this.children = new Human[0];
+        this.mother.setFamily(this);
+        this.father.setFamily(this);
+    }
+
+    public Family(Human[] children, Human father, Human mother, Pet pet) {
+        this.children = children;
+        this.father = father;
+        this.mother = mother;
+        this.pet = pet;
+    }
+
+    public Human[] getChildren() {
+        return children;
+    }
+
+    public void setChildren(Human[] children) {
+        this.children = children;
+    }
+
+    public Human getFather() {
+        return father;
+    }
+
+    public void setFather(Human father) {
+        this.father = father;
+        father.setFamily(this);
+    }
+
+    public Human getMother() {
+        return mother;
+    }
+
+    public void setMother(Human mother) {
+        this.mother = mother;
+        mother.setFamily(this);
+    }
+
+    public Pet getPet() {
+        return pet;
+    }
+
+    public void setPet(Pet pet) {
+        this.pet = pet;
+    }
+
+
+    public void addChild(Human child) {
+        Human[] newChildren = Arrays.copyOf(children, children.length + 1);
+        newChildren[children.length] = child;
+        setChildren(newChildren);
+    }
+
+    public boolean deleteChild(int index) {
+        if (index < 0 || index >= children.length) {
+            return false;
+        }
+
+        Human[] newChildren = new Human[children.length - 1];
+        for (int i = 0, j = 0; i < children.length; i++) {
+            if (i != index) {
+                newChildren[j++] = children[i];
+            }
+            children[i].setFamily(null);
```

Comment:
not here, pls move outside of the loop and use index field for setting family to null, in this case all child's family will be set null, which is not desired